components/ai_chat/core/browser/local_models_updater.cc

```diff
@@ -167,20 +165,12 @@ void ManageLocalModelsComponentRegistration(
   if (base::PathExists(GetDeprecatedComponentDir())) {
     base::DeletePathRecursively(GetDeprecatedComponentDir());
   }
-  if (!ai_chat::features::IsAIChatEnabled() ||
-      !ai_chat::features::IsPageContentRefineEnabled() || !cus) {
-    LocalModelsComponentInstallerPolicy::DeleteComponent();
-    return;
-  }
 
-  auto installer = base::MakeRefCounted<component_updater::ComponentInstaller>(
-      std::make_unique<LocalModelsComponentInstallerPolicy>());
-  installer->Register(
-      // After Register, run the callback with component id.
-      cus, base::BindOnce([]() {
-        brave_component_updater::BraveOnDemandUpdater::GetInstance()
-            ->EnsureInstalled(kComponentId);
-      }));
+  // TODO(fallaciousreasoning): Confirm this is the right approach.
+  // It seems like this was only being used for text embedding so we need to
+  // delete the component now?
+  // Maybe this whole thing should be deleted?
```

Comment:
cc @darkdh for what to do here?

Reply:
This whole file needs to be gone including where we call `ManageLeoLocalModelsComponentRegistration` to  register it on both desktop/Android and iOS and we need to delete the component folder if a user ever enabled the flag and got the component downloaded.

And don't forget to remove `test/data/leo/leo-local-models-updater/universal_sentence_encoder_qa_with_metadata.tflite` when we remove all the local_models_updater stuff including test.

Reply:
https://github.com/brave/brave-core/pull/28874/commits/846f06f9056e86f9413b8e405fece786fa4d6134

Not 100% sure if this is the best way to do it, but I got rid of everything here and added code to delete the downloaded component, but kept it in this file.

I didn't want to duplicate the deletion code in `brave/ios/browser/web/brave_web_main_parts.mm` and `brave/chromium_src/chrome/browser/component_updater/registration.cc` and this seemed like a sensible place to keep it. If there's a better way I'd rather do that though :laughing: